toolkit/components/uniffi-bindgen-gecko-js/src/render/cpp.rs

```diff
@@ -0,0 +1,135 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use askama::Template;
+use extend::ext;
+use heck::{CamelCase, MixedCase, SnakeCase};
+use uniffi_bindgen::interface::{ComponentInterface, FFIArgument, FFIFunction, FFIType};
+use super::shared::*;
+
+#[derive(Template)]
+#[template(path = "Scaffolding.cpp", escape = "none")]
+pub struct CPPScaffoldingTemplate {
+    pub ci: ComponentInterface,
+}
+
+#[derive(Template)]
+#[template(path = "Scaffolding.h", escape = "none")]
+pub struct CPPHeaderScaffoldingTemplate {
+    pub ci: ComponentInterface,
+}
+
+// Define extension traits with methods used in our template code
+
+#[ext(name=ComponentInterfaceCppExt)]
```

Comment:
TIL this works 😮 😮 This is really cool how we can use the names to differentiate between extensions